apis/v1beta1/types.go

```diff
@@ -167,6 +167,16 @@ type VirtualMachineCloneSpec struct {
 	// virtual machine is cloned.
 	// +optional
 	NumCoresPerSocket int32 `json:"numCoresPerSocket,omitempty"`
+	// CPUReservationMhz is the amount of CPU in MHz that is guaranteed available to the virtual machine.
+	// Defaults to the eponymous property value in the template from which the
+	// virtual machine is cloned.
+	// +optional
+	CPUReservationMhz int64 `json:"cpuReservationMhz,omitempty"`
+	// CPUShares are a relative priority to other virtual machines used in case of resource contention.
+	// Defaults to the eponymous property value in the template from which the
+	// virtual machine is cloned.
+	// +optional
+	CPUShares int32 `json:"cpuShares,omitempty"`
```

Comment:
Does it make sense to have something like:

```yaml
resources:
  reservation:
    cpu: ...
    memory: ...
  shares:
    ...
```

Or maybe this should be modelled in the k8s wordings which has `limits` and `requests`?:

This is e.g. done by the vm-operator APIs. However, vm-operator does not use shares.

Could someone research what the benefits are of setting shares? And should we also consider CPU Limit to be set?

Reply:
I'd prefer to takeover the definition how it is in vm-operator to have a similar API.

Which comes down to:

```
resources:
  requests: # --> reservations
    cpu: ...
    memory: ...
  limits: # --> limits
    cpu: ...
    memory: ...
```

And also use the types which are common for kubernetes.

https://github.com/vmware-tanzu/vm-operator/blob/main/api/v1alpha5/virtualmachineclass_types.go#L82-L86

The fields should have proper description to what this fields map at the end.

For shares: see my other comment.